### ISSUE TITLE
Mocking modules does not work with ansible-core 2.14

### ISSUE THREAD

**Issue by branic** [ansible/ansible-lint#2693]
##### Summary

ansible-lint with offline setting enabled is failing when used with ansible-core 2.14.0 if ansible-galaxy is not able to install collections.

##### Issue Type

- Bug Report

##### Ansible and Ansible Lint details

```console (paste below)
ansible --version

ansible [core 2.14.0]
  config file = /home/bevan185/git-repos/rosa-infrastructure/ansible.cfg
  configured module search path = ['/home/bevan185/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /home/bevan185/.local/lib/python3.9/site-packages/ansible
  ansible collection location = /home/bevan185/.ansible/collections:/usr/share/ansible/collections
  executable location = /usr/bin/ansible
  python version = 3.9.13 (main, Jun 15 2022, 11:24:50) [GCC 8.5.0 20210514 (Red Hat 8.5.0-13)] (/usr/bin/python3.9)
  jinja version = 3.0.3
  libyaml = True

ansible-lint --version

ansible-lint 6.8.6 using ansible 2.14.0
```

- ansible installation method: pip
- ansible-lint installation method: pip

##### OS / ENVIRONMENT

RHEL 8

##### STEPS TO REPRODUCE

1. Have a `collections/requirements.yml` that includes a collection that cannot be downloaded.

```yaml
---
collections:
  - name: fake_namespace.fake_collection
```

2. Configure `.ansible-lint` for offline mode and `mock_modules`

```yaml
offline: true

mock_modules:
  - fake_namespace.fake_collection.fake_module1
  - fake_namespace.fake_collection.fake_module2
```

3. Run ansible-lint the following error will be produced

```console
ERROR! Failed to resolve the requested dependencies map. Could not satisfy the following requirements:
* fake_namespace.fake_collection:* (direct request)

Got 1 exit code while running: ansible-galaxy collection install -v --offline -r collections/requirements.yml -p /home/username/.cache/ansible-compat/4ccf52/collections
```

##### Desired Behavior

ansible-lint should mock the modules in the collection so that ansible-galaxy install does not fail.

##### Actual Behavior

ansible-lint fails due to an ansible-galaxy collection install failure.

If ansible-lint is run with ansible 2.13.x ansible-galaxy is not run due to not supporting the --offline module and the modules are successfully mocked. This is the output from when using ansible 2.13.x

```console
WARNING Collection install skipped because ansible version before 2.14 do not support an offline mode.
```

**Comment by ssbarnea** [comment]
Fixed on main branch. I will make a release tomorrow morning as there are few others PRs I need to review and include.
